_posts/2015-01-19-enumerator-grep-vs-enumerator-select.markdown

```diff
@@ -0,0 +1,61 @@
+---
+layout: post
+title:  "Enumerable#grep vs Enumerable#select"
+date:   2015-04-09 11:37:00
+categories: ["ruby", "benchmark"]
+---
+
+Often, `Enumerable#select` is the chosen method to obtain elements from an
+Array for a given block. Without thinking twice, we may be doing more work than
+necessary by not taking advantage of another method from the Enumerable module,
+`Enumerable#grep`.
+
+`Enumerable#grep` can not always be used instead of `Enumerable#select`, but
+when it can, `#grep` provides better readability and a small speed improvement
+over `#select`.
+
+See the benchmark below for example, which uses the class matcher for `#grep`:
+
+```ruby
+2.1.2 :041 > a = (1..100000).map { rand(100000) }
+2.1.2 :042 > b = (1..100000).map { rand(100000).to_s }
+2.1.2 :043 > c = a + b
+2.1.2 :044 > Benchmark.bm(10) do |b|
+2.1.2 :045 >   b.report("select")    { c.select { |x| x.to_i == x } }
```

Comment:
Maybe you should  use `#is_a?` http://ruby-doc.org/core-2.2.1/Object.html#method-i-is_a-3F